lib/cisco_node_utils/vxlan_global.rb

```diff
@@ -0,0 +1,158 @@
+# VXLAN global provider class
+# Provides configuration of anycast gateways and duplicate host IP and
+# mac detection
+#
+# Alok Aggarwal, October 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node_util')
+
+module Cisco
+  # node_utils class for vxlan_global
+  class VxlanGlobal < NodeUtil
+    def feature_enable
+      config_set('vxlan_global', 'feature', state: '')
+    end
+
+    def feature_disable
+      config_set('vxlan_global', 'feature', state: 'no')
+    end
+
+    # Check current state of the configuration
+    def self.feature_enabled
```

Comment:
The naming convention we've followed for these is 'self.enabled' and 'enabled' (no need for 'disable' method, just use 'enable' for both)